integration/e2e_test.go

```diff
@@ -317,28 +363,108 @@ func deployMicrobot(t testing.TB) {
 				t.Log("Retrying")
 				time.Sleep(10 * time.Second)
 			}
+		} else {
+			if !strings.Contains(string(outputBytes), "calico") {
+				break
+			}
+		}
+	}
+	t.Log("No calico daemon set")
+}
+
+// installCilium installs cilium from helm chart
+func installCilium(t testing.TB) {
+	t.Log("Deploy Cilium")
+	command := []string{"helm", "install", "cilium", "--repo", "https://helm.cilium.io/",
+		"cilium", "--namespace", "kube-system", "--set", "cni.confPath=/var/snap/microk8s/current/args/cni-network",
+		"--set", "cni.binPath=/var/snap/microk8s/current/opt/cni/bin",
+		"--set", "daemon.runPath=/var/snap/microk8s/current/var/run/cilium",
+		"--set", "operator.replicas=1",
+		"--set", "ipam.operator.clusterPoolIPv4PodCIDRList=10.1.0.0/16",
+		"--set", "nodePort.enabled=true",
+	}
+
+	t.Logf("running command: %s", strings.Join(command, " "))
+	cmd := exec.Command(command[0], command[1:]...)
+	cmd.Env = append(cmd.Env, "KUBECONFIG="+KUBECONFIG)
+	outputBytes, err := cmd.CombinedOutput()
+	if err != nil {
+		t.Error(string(outputBytes))
+		t.Fatalf("Failed to deploy cilium from heml chart. %s", err)
+	}
+}
+
+// validateCilium checks a deployment of cilium daemon set.
+func validateCilium(t testing.TB) {
+	t.Log("Validate Cilium")
+
+	//check control plane machine exists
+	attempt := 0
+	maxAttempts := 120
+	machines := 0
+	command := []string{"kubectl", "get", "machine", "--no-headers"}
+	for {
+		t.Logf("running command: %s", strings.Join(command, " "))
+		cmd := exec.Command(command[0], command[1:]...)
+		outputBytes, err := cmd.CombinedOutput()
+		output := string(outputBytes)
+		if err != nil {
+			t.Log(output)
+			if attempt >= maxAttempts {
+				t.Fatal(err)
+			} else {
+				attempt++
+				t.Log("Retrying")
+				time.Sleep(10 * time.Second)
+			}
+		} else {
+			machines = strings.Count(output, "\n")
+			if machines > 0 {
+				break
+			}
+		}
+	}
+
+	t.Log("Checking Cilium daemon set")
+	command = []string{"kubectl", "--kubeconfig=" + KUBECONFIG, "-n", "kube-system", "wait", "ds/cilium", fmt.Sprintf("--for=jsonpath={.status.numberAvailable}=%d", machines)}
+	for {
```

Comment:
this retry is duplicated all over the place. Please refactor this logic into a separate function.